### PR TITLE
feat: static GameObjects implementation

### PR DESCRIPTION
**Pull request by Dedado3000**

### CODE REVIEW

File: Source/Modules/ModuleScene.h [hunk 1]
```diff
@@ -34,13 +34,21 @@ class ModuleScene : public Module
 	void OnPause();
 	void OnStop();
 
+	void AddGameObjectAndChildrens(GameObject* object);
```
Comment (gerard-queralt) [inline Source/Modules/ModuleScene.h:37]:
```suggestion
	void AddGameObjectAndChildren(GameObject* object);
```

File: Source/Modules/ModuleScene.h [hunk 2]
```diff
@@ -34,13 +34,21 @@ class ModuleScene : public Module
...
+	void RemoveGameObjectAndChildrens(GameObject* object);
```
Comment (gerard-queralt) [inline Source/Modules/ModuleScene.h:38]:
```suggestion
	void RemoveGameObjectAndChildren(GameObject* object);
```

File: Source/DataModels/Scene/Scene.h [hunk 3]
```diff
@@ -58,6 +58,7 @@ class Scene
 	const GameObject* GetAmbientLight() const;
 	const GameObject* GetDirectionalLight() const;
 	Quadtree* GetRootQuadtree() const;
+	const std::vector<GameObject*>& GetNonStaticObjects();
```
Comment (gerard-queralt) [inline Source/DataModels/Scene/Scene.h:61]:
```suggestion
	const std::vector<GameObject*>& GetNonStaticObjects() const;
```

File: Source/DataModels/Windows/EditorWindows/WindowInspector.cpp [hunk 4]
```diff
@@ -68,7 +68,14 @@ void WindowInspector::InspectSelectedGameObject()
 			{
 				lastSelectedGameObject->SetName(name);
 			}
-			
+			ImGui::SameLine();
+			bool staticness = lastSelectedGameObject->IsStatic();
+			if (ImGui::Checkbox("Static", &staticness))
+			{
+				lastSelectedGameObject->SetStatic(staticness);
+				lastSelectedGameObject->SpreadStatic();
```
Comment (gerard-queralt) [inline Source/DataModels/Windows/EditorWindows/WindowInspector.cpp:76]: I would add a comment here to remind us that we should update this behaviour, so it's not always propagated.

**Review by gerard-queralt — CHANGES_REQUESTED** [review]
Thanks for your work, David. Left some minor comments.

**Review by gerard-queralt — DISMISSED** [review]

File: Source/Modules/ModuleScene.cpp [hunk 5]
```diff
@@ -365,3 +356,51 @@ std::vector<GameObject*> ModuleScene::CreateHierarchyFromJson(Json& jsonGameObje
...
+		AddGameObjectAndChildren(children);
+	}
+
+}
+
+void ModuleScene::RemoveGameObjectAndChildren(GameObject* object)
+{
+	RemoveGameObject(object);
+
+	for (GameObject* children : object->GetChildren())
```
Comment (OriolSors) [inline Source/Modules/ModuleScene.cpp:376]:
```suggestion
	for (GameObject* child: object->GetChildren())
```
This is more readable

File: Source/Modules/ModuleScene.cpp [hunk 6]
```diff
@@ -365,3 +356,51 @@ std::vector<GameObject*> ModuleScene::CreateHierarchyFromJson(Json& jsonGameObje
...
+	{
+		RemoveGameObjectAndChildren(children);
```
Comment (OriolSors) [inline Source/Modules/ModuleScene.cpp:378]:
```suggestion
		RemoveGameObjectAndChildren(child);
```

File: Source/Modules/ModuleScene.h [hunk 7]
```diff
@@ -34,13 +34,21 @@ class ModuleScene : public Module
 	void OnPause();
 	void OnStop();
 
+	void AddGameObjectAndChildren(GameObject* object);
+	void RemoveGameObjectAndChildren(GameObject* object);
+
```
Comment (OriolSors) [inline Source/Modules/ModuleScene.h:39]: Did you delete these functions from the Quadtree?

**Review by OriolSors — COMMENTED** [review]
Good implementation and ready for simplify the Quadtree @Dedado3000 !

Please, take a look what I commented about deleting the redundant functions in the Quadtree with your new implementation. Maybe we can delete that with the refactor.

Accept please my suggestions, its more clear xd

**Review by OriolSors — APPROVED** [review]

**Review by gerard-queralt — APPROVED** [review]

**Pull request merged**